ENH: Changes the content review email template from a textareafield to an HTMLEditorField

Radroaches opened the pull request:
Currently users have to enter their own HTML tags into the editor if they want to style the review email. 

This would also allow users to preview the HTML before sending the email.

Radroaches committed:
ENH: Content review email template update

Changed the content review email template from a Textareafield to an HTMLEditorField
with a custom TinyMCEConfig to remove options that won't work

Radroaches committed:
Merge branch 'silverstripe:4' into pulls/4/content_review_template_update

Radroaches committed:
Merge branch 'silverstripe:4' into pulls/4/content_review_template_update

ChloeSartorelli commented on src/Extensions/ContentReviewDefaultSettings.php:
```diff
@@ -164,13 +165,51 @@ public function updateCMSFields(FieldList $fields)
+     *
+     * @return TinyMCEConfig
+     */
+    private function getTinyMCEConfig(
+        TinyMCEConfig $config
+    ): TinyMCEConfig {
+        $editorButtonsGroupSeparator = '|';
+        $allowedEditorButtons = [
+            'undo',
+            'redo',
+            $editorButtonsGroupSeparator,
+            'bold',
+            'italic',
+            'underline',
+            $editorButtonsGroupSeparator,
+            'bullist',
+            'numlist',
+            $editorButtonsGroupSeparator,
+            'sslink',
+            $editorButtonsGroupSeparator,
+            'formatselect',
+            $editorButtonsGroupSeparator,
+            'code',
+        ];
+
+        $config->setButtonsForLine(1, $allowedEditorButtons);
+        $config->setButtonsForLine(2, []);
+        $config->setButtonsForLine(3, []);
+
+        return $config;
```
The config should be defined in `_config.php` and be given its own name - this allows developers to easily modify it in their own projects however they may so choose. Also there's no need to set a variable for `|`.

jareddreyerss replied:
@GuySartorelli removed the tinymce helper as we can put this in our projects as agreed doesn't need to live in this module.

GuySartorelli reviewed (changes requested):
My main concern with this is that somebody may have some html set in this field which will now be stripped out by the tinymce config, which makes this a breaking change.

jareddreyerss committed:
Removed custom tinymce helper

Was not required because SilverStripe can allow the developer to set their own `TinyMCEConfig::class` with any endpoint they see fit e.g. DataExtensions .

This helps negate overriding configs on existing projects and means we do not need to force custom tinymce configs.

GuySartorelli commented:
Sorry it's taken me so long to come back to this one.
Is there an easy way to define a specific config for that one `HTMLEditorField` instance? Or do we need a `private static string $content_review_body_config_name = '';` or similar, together with something like this?
```php
$configName = $this->owner->config()->get('content_review_body_config_name');
if ($configName) {
    $wysiwyg->setEditorConfig(HTMLEditorConfig::get($configName));
}
```

jareddreyerss commented:
> Sorry it's taken me so long to come back to this one. Is there an easy way to define a specific config for that one `HTMLEditorField` instance? Or do we need a `private static string $content_review_body_config_name = '';` or similar, together with something like this?
> 
> ```
> $configName = $this->owner->config()->get('content_review_body_config_name');
> if ($configName) {
>     $wysiwyg->setEditorConfig(HTMLEditorConfig::get($configName));
> }
> ```

😄  just thinking now... we did not actually need a PR for this change at all 🥴 . 

What would you guys like? am happy to put in a config member as above so it's easily switchable and doesn't break current projects.  Otherwise I am also happy to close this PR since its a small change and can be done by the developer via project code.

GuySartorelli commented:
Nobody else seems to be clamoring for this to be merged - I'm not fussed either way tbh :p
If you're happy to resolve this in projects and close the PR then that's absolutely fine. If someone else wants this functionality in the future they're welcome to resurrect this PR.

GuySartorelli commented:
Closing as there was no response since my last comment.
